lib/cookie.c

```diff
@@ -779,10 +779,16 @@ Curl_cookie_add(struct Curl_easy *data,
       offt = curlx_strtoofft((*co->maxage == '\"')?
                              &co->maxage[1]:&co->maxage[0], NULL, 10,
                              &co->expires);
-      if(offt == CURL_OFFT_FLOW)
+      switch(offt) {
+      case CURL_OFFT_FLOW:
         /* overflow, used max value */
         co->expires = CURL_OFF_T_MAX;
-      else if(!offt) {
+        break;
+      case CURL_OFFT_INVAL:
+        /* negative or otherwise bad, expire */
+        co->expires = 1;
```

Comment:
It was unexpected to me that curlx_strtoofft does not parse negative numbers, what is the reason for that?

Reply:
I can't recall the exact reasoning anymore, but I did look over all current uses of this function and I could not spot any other code path that would expect a negative value to pass through fine.

Reply:
#9934. Also I notice MS doc for _strtoi64 says it does not set ERANGE on overflow/underflow so I will investigate further.

Reply:
> I could not spot any other code path that would expect a negative value to pass through fine.

hm appears I misread..